### DISCUSSION TITLE
Tracked moisture: nan%.

### DISCUSSION THREAD

**Discussion by hidabril** [WAM2layers/WAM2layers#400] [Q&A] [answered]
Hi,

I was trying to use wam2layers, but I am getting NaN values in the tracking part. 
![image](https://github.com/WAM2layers/WAM2layers/assets/125105779/c6826a0f-cf1a-44fc-8776-943a3f742acf)
![codigo](https://github.com/WAM2layers/WAM2layers/assets/125105779/40255982-abbc-48f6-9f37-ff412c80cccc)

I tried downloading a smaller area instead of the whole world, but I still get the error:

ValueError: dimensions ('level', 'latitude', 'longitude') must have the same length as the number of data dimensions, ndim=4

I don’t know what to do in either situation.

**Comment by ruudvdent** [comment]
Dear Hidabril, welcome to WAM2layers! Please give a small introduction of yourself to the community in #206 

Seems you are talking about 2 separate problems, correct?

Regarding the nans. I suspect something went wrong in the preprocessing already. Are there nans in there already? What do your preprocessed files look like compared to the example cases?

The error "ValueError: dimensions ('level', 'latitude', 'longitude') must have the same length as the number of data dimensions, ndim=4" occurs where exactly?

Note that we made a v3.1.0 release last week. Do your errors occurs with that version?

**Reply by BSchilperoort** [reply]
> Regarding the nans. I suspect something went wrong in the preprocessing already. Are there nans in there already? What do your preprocessed files look like compared to the example cases?

Alternatively, it can be that no moisture is being tagged. The statistics are computed like this:
```py
tracked_percentage = tracked / total_tagged_moisture * 100
```
So if `total_tagged_moisture` is 0, the result will be NaN as the code tries to divide by 0.

**Reply by ruudvdent** [reply, marked as answer]
Good point! So this could possibly relate to evaporation (in forward mode) or precipitation (in backward mode) being zero in the tagging region. Or the tagging region is itself is somehow not correctly implemented, but I don't see any obvious errors in the config file above.
